components/playlist/browser/playlist_service.h

```diff
@@ -215,6 +215,15 @@ class PlaylistService : public KeyedService,
 
   bool HasPlaylistItem(const std::string& id) const;
 
+  // Returns true when we should try getting media from a background web
+  // contents, which means it could have impact on performance/memory.
+  bool ShouldGetMediaFromBackgroundWebContents(const GURL& url) const;
+
+  // Returns true if the url is known to have media file. As this is decided
+  // based on a given list and url format, it could be used when
+  // ShouldGetMediaFromBackgroundWebContents() returns true.
+  bool CouldURLHaveMedia(const GURL& url);
```

Comment:
I like how the `CouldURLHaveMedia` name implies that it won't definitely have media. However, when this function returns `false` it implies that the URL cannot have media, which is not neccessarily the case. Perhaps a more truthful name would be something like `ShouldURLHaveMedia` since we're looking up a list of known URL patterns with media?